client/db/src/bench.rs

```diff
@@ -525,6 +505,40 @@ impl<B: BlockT> StateBackend<HashFor<B>> for BenchmarkingState<B> {
 		*self.whitelist.borrow_mut() = new;
 	}
 
+	fn get_read_and_written_keys(&self) -> Vec<(Vec<u8>, u32, u32, bool)> {
+		// We only track at the level of a key-prefix and not whitelisted for now for memory size.
+		// TODO: Refactor to enable full storage key transparency, where we can remove the
+		// `prefix_key_tracker`.
+		let mut prefix_key_tracker = HashMap::<[u8; 32], (u32, u32, bool)>::new();
+		self.all_trackers().iter().for_each(|tracker| {
+			if !tracker.whitelisted {
+				let mut prefix = [0u8; 32];
+				prefix[0..32].copy_from_slice(&tracker.key[0..32]);
```

Comment:
this is panicking for some specific keys like `:code` or `:heappage`, maybe we shouldn't make the assumption that the prefix is 32 byte, and instead consider it can be of any length.

Thus system pallet could declare `:heappage` prefix as in the storage_info:
```
StorageInfo {
pallet_name: b"System",
storage_name: b"heappage well known key",
prefix: ":heappage",
max_values: Some(1),
max_size: ...,
}
```
And it would be a good information about heappage storage.

Reply:
actually instead of pallet_name and storage_name fields maybe we can merge them together as `desc` field which will be a description of the storage

Reply:
ah good catch. yeah probably we should update the format here. Using a vec would solve our issues?

Reply:
yes a vec here should solve it